components/brave_shields/core/browser/ad_block_filter_list_catalog_provider.cc

```diff
@@ -11,13 +11,30 @@
 #include "base/files/file_path.h"
 #include "base/task/thread_pool.h"
 #include "brave/components/brave_shields/core/browser/ad_block_component_installer.h"
+#include "brave/components/brave_shields/resources/grit/adblock_bundle.h"
+#include "components/component_updater/component_updater_service.h"
+#include "ui/base/resource/resource_bundle.h"
 
 constexpr char kListCatalogFile[] = "list_catalog.json";
 
+namespace {
+
+std::string LoadBundledResource() {
+  return ui::ResourceBundle::GetSharedInstance().LoadDataResourceString(
+      IDR_BRAVE_ADBLOCK_LIST_CATALOG);
```

Comment:
Frankly speaking, the list looks heavy (84KB JSON file). 
1. Let's add preprocess the list or during the build. Some fields could be easily dropped.
2. I suggest to drop `base64_public_key`, especially we going to stop download it. The sec review: https://github.com/brave/reviews/issues/1175#event-8376738786
3. If it's not enough, we could repack it to protobuf/flatbuf to eliminate parsing completely.
4. Are we going to remove the component `Brave Ad Block List Catalog`? Otherwise, we're starting to do double work each start.

Reply:
> Are we going to remove the component `Brave Ad Block List Catalog`? Otherwise, we're starting to do double work each start.

This PR is now using Chromium's existing preinstalled component code, so double parsing is no longer an issue.